Nos/Models/Author+CoreDataClass.swift

```diff
@@ -194,6 +194,46 @@ public class Author: NosManagedObject {
         )
         return fetchRequest
     }
+    
+    @nonobjc func lookupReportsByAuthor() -> NSFetchRequest<Event> {
+        let fetchRequest = NSFetchRequest<Event>(entityName: "Event")
+        fetchRequest.sortDescriptors = [NSSortDescriptor(keyPath: \Event.createdAt, ascending: false)]
+        fetchRequest.predicate = NSPredicate(
+            format: "kind = %i AND author = %@",
+            EventKind.report.rawValue,
+            self
+        )
+        return fetchRequest
+    }
+    
+    @nonobjc func lookupReportsOnAuthor(context: NSManagedObjectContext) -> [Event] {
+        let fetchRequest = NSFetchRequest<Event>(entityName: "Event")
+        fetchRequest.sortDescriptors = [NSSortDescriptor(keyPath: \Event.createdAt, ascending: false)]
+
+        // Unwrap hexadecimalPublicKey or provide a default value
+        let publicKey = self.hexadecimalPublicKey ?? ""
+
+        // Create a predicate for the "tags" condition
+        let tagsPredicate = NSPredicate(format: "SUBQUERY(allTags, $tag, $tag.p == %@ AND $tag.tagValue == %@).@count > 0", "p", publicKey)
+
+        // Set the predicate for the fetch request
+        fetchRequest.predicate = tagsPredicate
+            
+        var allEvents: [Event] = []
+        do {
+            allEvents = try context.fetch(fetchRequest)
+        } catch {
+            print("Failed to fetch events. Error: \(error.localizedDescription)")
+        }
+
+        // Filter the events to only include those of kind 'report'
+        let reportEvents = allEvents.filter {
```

Comment:
It's much faster to let the database filter the `kind` than to do it in memory here. You can add `AND kind = %@` to the predicate above (and then add EventKind.report.rawValue to the varargs).